src/Simulation/qdk_sim_rs/src/processes/apply.rs

```diff
@@ -244,3 +340,600 @@ pub(crate) fn apply_pauli_channel(
+                        c64!(-0.026517676231910946-0.050044762307278966 i)
+                    ],
+                    [
+                        c64!(-0.03871603769659798+0.07306578778099337 i),
+                        c64!(0.08445493432837799-0.010382442060839553 i),
+                        c64!(-0.006610521592267533+0.0024834722807521113 i),
+                        c64!(0.005989905700018601+0.00411411762104997 i),
+                        c64!(0.025365377625872418),
+                        c64!(-0.014944103105233367-0.02140064303597717 i),
+                        c64!(0.0016225956405177121+0.001435116293025267 i),
+                        c64!(0.00025484244892756214-0.0022144797831879417 i)
+                    ],
+                    [
+                        c64!(-0.038835549693321744-0.0757114992294755 i),
+                        c64!(-0.040997312372063874+0.07737106128583843 i),
+                        c64!(0.001799319259396504-0.00704041080461318 i),
+                        c64!(-0.007000034992844309+0.0026298065328700434 i),
+                        c64!(-0.014944103105233367+0.02140064303597717 i),
+                        c64!(0.026859988052306022),
+                        c64!(-0.0021667585187491217+0.0005234719718873384 i),
+                        c64!(0.0017182042452060908+0.0015196780057005993 i)
+                    ],
+                    [
+                        c64!(0.0016572758798486477+0.006864405797239211 i),
+                        c64!(0.004815075032589164-0.005442424691760617 i),
+                        c64!(-0.025042113573289376+0.047260046864100305 i),
+                        c64!(0.05462671758535295-0.0067155191678741144 i),
+                        c64!(0.001622595640517712-0.0014351162930252683 i),
+                        c64!(-0.002166758518749123-0.0005234719718873377 i),
+                        c64!(0.016406706500140797),
+                        c64!(-0.009666069915131952-0.013842256732161828 i)
+                    ],
+                    [
+                        c64!(-0.006767855076913282-0.002645953845562754 i),
+                        c64!(0.0017549279568661097+0.007268878879688429 i),
+                        c64!(-0.025119415724371576-0.04897133269077113 i),
+                        c64!(-0.026517676231910956+0.050044762307278966 i),
+                        c64!(0.00025484244892756344+0.002214479783187942 i),
+                        c64!(0.00171820424520609-0.0015196780057006006 i),
+                        c64!(-0.009666069915131954+0.013842256732161827 i),
+                        c64!(0.017373442929624686)
+                    ]
+                ];
+                for (actual, expected) in actual.iter().zip(expected.iter()) {
+                    assert_abs_diff_eq!(actual, expected, epsilon = 1e-6);
+                }
+                Ok(())
+            }
+            _ => panic!("did not return density operator as expected"),
```

Comment:
The fragment `"did not return density operator as expected"` occurs more than once. Consider defining it as a global constant or a macro. Or event the whole fragment `panic!("did not return density operator as expected")` as a (shorter) macro.

Reply:
That's OK.